packages/cli/src/commands/deploy/service.ts

```diff
@@ -43,15 +45,94 @@ export default class Service extends Command {
       },
       {
         title: 'Creating service',
-        skip: () => 'not yet implemented',
-        task: async () => {
-          // service = await create(lcd, definition, credential.mnemonic)
+        task: async (ctx, task) => {
+          const title = task.title
+          const logs = await this.deploy('service', definition, user.uid)
+          logs.on('data', x => {
+            task.title = `${title} (${x.message})`
+          })
+          service = await this.toPromise(logs)
         }
-      },
+      }
     ])
     await tasks.run()
 
-    // this.log(`Service deployed with the hash ${service.hash}`)
-    // this.log(`https://explorer.testnet.mesg.com/services/${service.hash}`)
+    this.log('')
+    this.log(`Service deployed with the hash ${service.hash}`)
+    this.log('')
+
+    const { deploy } = await prompt({ type: 'confirm', name: 'deploy', message: 'Would you like to start this service?' })
+
+    if (deploy) {
+      const env = await this.getEnv(definition.configuration.env || [])
+
+      const logs = await this.deploy('runner', {
+        serviceHash: service.hash,
+        env
+      }, user.uid)
+      logs.on('data', this.log)
+
+      const runner = await this.toPromise(logs)
+      this.log('')
+      this.log(`Service started with the runner hash ${runner.hash}`)
+    }
+
+    return firebase.firestore().terminate()
+  }
+
+  async getEnv(envs: string[]): Promise<string[]> {
+    if (!envs.length) return []
+
+    const res = await prompt(envs.map((x: string) => ({
+      name: x.split('=')[0],
+      default: x.split('=')[1],
+      type: 'input'
+    }))) as { [key: string]: string }
+    return Object.keys(res).map(x => `${x}=${res[x]}`)
+  }
+
+  private async deploy(type: 'service' | 'runner', definition: any, uid: string): Promise<EventEmitter> {
+    const res = await firebase.firestore().collection('deployments').add({ type, uid, definition })
+    const eventEmitter = new EventEmitter()
+
+    let unsubscribeLogs: () => void
+    let unsubscribeDeployment: () => void
+
+    const clearListeners = () => {
+      if (unsubscribeDeployment) unsubscribeDeployment()
+      if (unsubscribeLogs) unsubscribeLogs()
+    }
+
+    unsubscribeLogs = res.collection('logs').onSnapshot(
+      snapshots => {
+        for (const change of snapshots.docChanges()) {
+          if (change.type !== 'added') continue
+          const data = change.doc.data()
+          if (data.level === 'error') return eventEmitter.emit('error', new Error(data.message))
+          eventEmitter.emit('data', data)
+        }
+      },
+      error => eventEmitter.emit('error', error)
+    )
+
+    unsubscribeDeployment = res.onSnapshot(
+      async snapshot => {
+        const data = snapshot.data()
+        if (!data.resourceRef) return
```

Comment:
shouldn't it return an error?

Reply:
no this is if there is an update on the document that is not related. Here we only want updates that add the `resourceRef` meaning that the service is created